src/main/java/org/hyperledger/fabric/sdk/Channel.java

```diff
@@ -1147,7 +1147,7 @@ public PeerOptions getPeersOptions(Peer peer) {
      * @return old options.
      */
 
-    PeerOptions setPeerOptions(Peer peer, PeerOptions peerOptions) throws InvalidArgumentException {
+    public PeerOptions setPeerOptions(Peer peer, PeerOptions peerOptions) throws InvalidArgumentException {
```

Comment:
Given that the peer options is a way to let the SDK request for private data. If one uses the SDK, they should have an ability to set the option. Is there a concern with this exposure?

Alternatives:
1. Consider a new method to set the peer's block request type option explicitly.
2. Consider a global setting at the channel level (instead of a peer option), to request generically private data from all the peers.
Let me know your thoughts.

Reply:
I’m not sure it is the way I would have designed the API but the suggested usage pattern can be seen here:

- https://github.com/hyperledger/fabric-sdk-java/blob/9c4c3aa075b8c5673dc997721e51db2dfd70d024/src/test/java/org/hyperledger/fabric/sdkintegration/End2endAndBackAgainIT.java#L651
- https://github.com/hyperledger/fabric-sdk-java/blob/9c4c3aa075b8c5673dc997721e51db2dfd70d024/src/test/java/org/hyperledger/fabric/sdkintegration/UpdateChannelIT.java#L489

Reply:
Thanks for the test case references. Otherwise, I was heading in the direction to reuse the channel instance created using the gateway SDK.